client/src/components/ResultPanel/ResultPanelSubscriptionProvider.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright © 2024, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { Disposable, Uri, commands, window, workspace } from "vscode";
+
+import { SubscriptionProvider } from "../SubscriptionProvider";
+import { resultPanelManager } from "./ResultPanelManager";
+
+const scriptRegex = /<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi;
+
+interface ResultPanelContext {
+  webview: string;
+  resultPanelId: string;
+}
+export class ResultPanelSubscriptionProvider implements SubscriptionProvider {
+  getSubscriptions(): Disposable[] {
+    return [
+      commands.registerCommand(
+        "SAS.saveHTML",
+        async (context: ResultPanelContext) => {
+          const panel =
```

Comment:
Since we really only care about the panel html, how about making a method in `resultPanelManager` to give us back the html? That way, the logic removing the `<script>` tags can be encapsulated in the same spot where they're added

Reply:
Refactored in https://github.com/sassoftware/vscode-sas-extension/pull/935/commits/fcfa9ddc0204a6fde9e630c22203f1e99f85e357. Note that I dont strip the html on panel hydration, because I still want the setstate script in there so that we can support downloading of rehydrated webview panels.